### PR TITLE
Related calldata and concretion improvements/fixes

### PR DESCRIPTION
**Pull request by nbanmp**
Includes:
Removing `ForAll` - Faster!
Minimizing `calldatasize` - No crazy long transactions.
Bugfixes

### CODE REVIEW

File: mythril/laser/ethereum/state.py [hunk 1]
```diff
@@ -56,28 +57,29 @@ def __init__(self, tx_id, starting_calldata=None):
             self.calldatasize = BitVec("{}_calldatasize".format(self.tx_id), 256)
             self.concrete = False
 
-        self.starting_calldata = starting_calldata or []
```
Comment (norhh) [inline mythril/laser/ethereum/state.py]: I think this should be reflected in `native_contracts.py`

Comment (nbanmp) [reply mythril/laser/ethereum/state.py]: All done, the modifications are reflected in `natives.py` now.

File: mythril/analysis/solver.py [hunk 2]
```diff
@@ -62,6 +67,7 @@ def get_transaction_sequence(global_state, constraints):
     txs = {}
```
Comment (JoranHonig) [inline mythril/analysis/solver.py]: Its not part of this pr, but I was wondering why the model used in the vulnerability check module isn't reused here

File: mythril/laser/ethereum/instructions.py [hunk 3]
```diff
@@ -504,6 +505,8 @@ def calldatacopy_(self, global_state):
                         i_data + 1 if isinstance(i_data, int) else simplify(i_data + 1)
                     )
 
+                    environment.calldata.update_constraints(state)
```
Comment (JoranHonig) [inline mythril/laser/ethereum/instructions.py]: Why this? Didn't this already happen in the transactions initial state?

File: mythril/laser/ethereum/state.py [hunk 4]
```diff
@@ -111,6 +113,9 @@ def __getitem__(self, item):
             except IndexError:
                 return BitVecVal(0, 8)
         else:
+            constraints = [Implies(item != 0, UGT(self.calldatasize, item))]
```
Comment (JoranHonig) [inline mythril/laser/ethereum/state.py]: Shouldn't it be:
```suggestion
            constraints = [Implies(self._calldata[item] != 0, UGT(self.calldatasize, item))]
```
